File: examples/src/Shapes/SubstitutePictureTitleOfOLEObjectFrame.cs
import aspose.slides as slides
using Aspose.slides.Examples.CSharp
using System
using System.Collections.Generic
import aspose.pydrawing as drawing
using System.IO
using System.Linq
using System.text
using System.Threading.Tasks
using Aspose.slides.DOM.Ole

namespace CSharp.shapes
{
    class SubstitutePictureTitleOfOLEObjectFrame
    {
        public static void Run() {

            #ExStart:SubstitutePictureTitleOfOLEObjectFrame
            # The path to the documents directory.
            dataDir = RunExamples.GetDataDir_Shapes()
            oleSourceFile = dataDir +"ExcelObject.xlsx"
            oleIconFile = dataDir + "Image.png"

            with slides.Presentation() as pres:
            {
                image = None
                slide = pres.slides[0]

                # Add Ole objects
                byte[] allbytes = File.ReadAllBytes(oleSourceFile)
                IOleEmbeddedDataInfo dataInfo = new OleEmbeddedDataInfo(allbytes, "xlsx")
                IOleObjectFrame oof = slide.shapes.AddOleObjectFrame(20, 20, 50, 50, dataInfo)
                oof.IsObjectIcon = True

                # Add image object
                byte[] imgBuf = File.ReadAllBytes(oleIconFile)
                using (MemoryStream ms = new MemoryStream(imgBuf))
                {
                    image = pres.images.add_image(drawing.Bitmap(ms))
                }
                oof.substitute_picture_format.picture.image = image

                # Set caption to OLE icon
                oof.SubstitutePictureTitle = "Caption example"
            }

            #ExEnd:SubstitutePictureTitleOfOLEObjectFrame

        }
    }
}

File: examples/src/shapes/substitute_picture_title_of_ole_object_frame.py
﻿import aspose.slides as slides


def substitute_picture_title_of_ole_object_frame(global_opts):
    with slides.Presentation() as pres:
        slide = pres.slides[0]

        # Add Ole objects
        with open(global_opts.data_dir + "book.xlsx", "rb") as file:
            all_bytes = file.read()

        data_info = slides.dom.ole.OleEmbeddedDataInfo(all_bytes, "xlsx")

        oof = slide.shapes.add_ole_object_frame(20, 20, 50, 50, data_info)
        oof.is_object_icon = True

        # Add image object
        img = slides.Images.from_file(global_opts.data_dir + "image1.jpg")
        imgx = pres.images.add_image(img)
        oof.substitute_picture_format.picture.image = imgx

        # Set caption to OLE icon
        oof.substitute_picture_title = "Caption example"
